libraries/Microsoft.Bot.Connector.Streaming/Application/IStreamingConnectionFactory.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Net.WebSockets;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Bot.Connector.Streaming.Application
+{
+    /// <summary>
+    /// A factory that can create an instance of <see cref="StreamingConnection"/>.
+    /// </summary>
+    public interface IStreamingConnectionFactory
```

Comment:
Some questions around this. Can we avoid defining more public types, such as IStreamingConnectionFactory and StreamingConnectionFactory? Originally the idea is that the only new type to allow different streaming strategies is StreamingConnection. Can't we use streaming connection to inject different connections and then test the connections in isolation? What value does this add if that is the case?

Reply:
The main issue here is that we are only enabling both new and old streaming library to exist side-by-side as different transport strategies for a limited amount of time while we transition to the new code. So we would not want to expose this to only remove it in a few releases.